modules/example/src/smithy4s/example/OrderType.scala

```diff
@@ -0,0 +1,41 @@
+package smithy4s.example
+
+import smithy4s.schema.Schema._
+
+sealed trait OrderType extends scala.Product with scala.Serializable
+object OrderType extends smithy4s.ShapeTag.Companion[OrderType] {
+  val id: smithy4s.ShapeId = smithy4s.ShapeId("smithy4s.example", "OrderType")
+
+  val hints : smithy4s.Hints = smithy4s.Hints.empty
+
+  case class OnlineCase(online: OrderNumber) extends OrderType
```

Comment:
following stephen's question, could we get rid of the `Case` part?

Reply:
I'm opposed to this, until we have a better mechanism for collision detection in place.